src/routing_node.rs

```diff
@@ -1039,8 +1039,12 @@ impl<F> RoutingNode<F> where F: Interface {
             return true;
         }
 
-        let close_group = self.routing_table.our_close_group();
-        closer_to_target(&address, &self.routing_table.our_close_group().pop().unwrap().id(), &self.own_name)
+        match self.routing_table.our_close_group().pop() {
+            Some(furthest_close_node) => {
+                closer_to_target_or_equal(&address, &furthest_close_node.id(), &self.own_name)
```

Comment:
think the or_equal part may be where this is happening so multiple calls from same node is not closer but may be equal ? May be worth checking here with routing_table.check_node() call that we do not already have the node perhaps?